src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4365,10 +4370,20 @@ public String descriptorString() {
         } else if (isHidden()) {
             String name = getName();
             int index = name.indexOf('/');
-            return "L" + name.substring(0, index).replace('.', '/')
-                       + "." + name.substring(index+1) + ";";
+            return new StringBuilder(name.length() + 2)
+                    .append('L')
+                    .append(name.substring(0, index).replace('.', '/'))
+                    .append('.')
+                    .append(name.substring(index + 1))
```

Comment:
`.append(name, index + 1, name.length())` might be a small win here, but it might be hard to benchmark this branch since it's only for hidden classes.

Reply:
Indeed, I've measured the case for hidden classes with benchmark
```java
@State(Scope.Thread)
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.NANOSECONDS)
public class HiddenClassBenchmark {
  private Class<?> hiddenClass;

  @Setup
  public void setUp() throws Exception {
    byte[] bytes = getClassWriter().toByteArray();

    hiddenClass = MethodHandles
            .lookup()
            .defineHiddenClass(bytes, true, NESTMATE)
            .lookupClass();

    if (hiddenClass.isHidden()) {
      return;
    }
    throw new RuntimeException();
  }

  @Benchmark
  public String descriptorString() {
    return hiddenClass.descriptorString();
  }

  private static ClassWriter getClassWriter() {
    ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_MAXS);

    var name = HiddenClassDemo.class.getName().replace('.', '/');
    cw.visit(V1_6, ACC_PUBLIC + ACC_SUPER, name, null, "java/lang/Object", null);
    cw.visitEnd();
    return cw;
  }

  private static class HiddenClassDemo {
  }
}
```
and got those results:
```
jdk 16
Benchmark                                                               Mode  Cnt     Score    Error   Units
HiddenClassBenchmark.descriptorString                                   avgt  100   112.591 ±  1.320   ns/op
HiddenClassBenchmark.descriptorString:·gc.alloc.rate.norm               avgt  100   600.045 ±  0.001    B/op

patched

Benchmark                                                               Mode  Cnt     Score    Error   Units
HiddenClassBenchmark.descriptorString                                   avgt  100    85.958 ±  0.561   ns/op
HiddenClassBenchmark.descriptorString:·gc.alloc.rate                    avgt  100  3312.293 ± 20.571  MB/sec
HiddenClassBenchmark.descriptorString:·gc.alloc.rate.norm               avgt  100   448.034 ±  0.001    B/op

patched without substring

Benchmark                                                               Mode  Cnt     Score    Error   Units
HiddenClassBenchmark.descriptorString                                   avgt  100    76.580 ±  0.587   ns/op
HiddenClassBenchmark.descriptorString:·gc.alloc.rate                    avgt  100  3585.818 ± 26.933  MB/sec
HiddenClassBenchmark.descriptorString:·gc.alloc.rate.norm               avgt  100   432.031 ±  0.001    B/op
```